mathesar/templatetags/client_files.py

```diff
@@ -0,0 +1,122 @@
+import logging
+import json
+import os.path
+from urllib.parse import urljoin
+from django import template
+from django.conf import settings
+from django.utils.safestring import mark_safe
+
+
+logger = logging.getLogger(__name__)
+register = template.Library()
+
+MODE = getattr(settings, "MODE")
+CLIENT_BUILD_LOCATION = getattr(settings, "CLIENT_BUILD_LOCATION")
+MANIFEST = os.path.join(CLIENT_BUILD_LOCATION, "manifest.json")
+
+STATIC_URL = getattr(settings, "STATIC_URL")
+CLIENT_DEV_URL = getattr(settings, "CLIENT_DEV_URL")
+
+
+class ClientAssetHandler:
```

Comment:
Could you add some comments to this method that explain what it's doing for people that are not familiar with frontend code structure?

Reply:
I will add comments and tests for this file.

Reply:
Added comments and tests. Please let me if the test approach is okay.

The file loads the settings when the module loads, and the defined class is a singleton which stores the result. So, for each individual test, I've reloaded the entire file, inorder to reload the settings (which change per test), and to try to read the manifest.json file for each test.